### ISSUE TITLE
Configurable Preset Grid Filters

### ISSUE THREAD

**Issue by geographika** [compassinformatics/cpsi-mapview#401]
Allow the `CpsiMapview.controller.grid.Grid` class to have configured pre-set filters. 
A new flag could also be set to clear all filters, or to reset to the pre-set filters. 

For simple filters, for example setting a year equal to a value the following works:
```js
    applyGridFilter: function (gridWindow, columnName, value) {

        var grid = gridWindow.down('grid');
        var column = grid.columnManager.getHeaderByDataIndex(columnName);
        if (!column.filter.value) {
            column.filter.value = value;
        }
    },

   // to call
   me.applyGridFilter(gridWindow, 'YearOfWorks', { eq: currentyear });
```

However list filters should also be supported which seems trickier as the filter menus need to be kept in synch:

```
        var menu = grid.columnManager.headerCt.getMenu();
        var menuitem = menu.items.get('columnItem');
        var columnMenu = menuitem.getMenu();
```

The current demo does not currently have any list filters, so this would need to be setup for testing. 

![image](https://user-images.githubusercontent.com/490840/117658673-4cb66080-b19b-11eb-90ed-cb82eb835a34.png)

The grid filters should be configurable from the [config file](https://github.com/compassinformatics/cpsi-mapview/blob/master/resources/data/layers/default.json)
The JSON config objects could match the [Ext.util.Filter](https://docs.sencha.com/extjs/6.7.0/classic/Ext.util.Filter.html) structures as much as possible:

```json

"gridFilters": [{
    "property": "age",
    "value": 42,
    "operator": "<"
}]
```

Unit tests would also be useful for this new feature.

**Comment by JakobMiksch** [comment]
@geographika do you have layer which I can use for testing the `list` filter?
I tried `tourist_info`, but in contrast to `ruins` CPSI cannot seem to display it as grid. This is the error from the Server:
```
<?xml version="1.0" encoding="UTF-8"?>
<ows:ExceptionReport xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:ows="http://www.opengis.net/ows/1.1" version="2.0.0" xml:lang="en-US" xsi:schemaLocation="http://www.opengis.net/ows/1.1 http://schemas.opengis.net/ows/1.1.0/owsExceptionReport.xsd">
  <ows:Exception exceptionCode="OperationProcessingFailed" locator="mapserv">
    <ows:ExceptionText>msWFSGetFeature(): WFS server error. Layer tourist_info does not support sorting</ows:ExceptionText>
  </ows:Exception>
</ows:ExceptionReport>

```

**Comment by JakobMiksch** [comment]
@geographika regarding setting the list filter:

you mentioned this code snippet:

```js
 var menu = grid.columnManager.headerCt.getMenu();
 var menuitem = menu.items.get('columnItem');
 var columnMenu = menuitem.getMenu();
```

is it intended to use this snippet to set the filter value or is it only for syncing the filter menu?

**Comment by geographika** [comment]
> is it intended to use this snippet to set the filter value or is it only for syncing the filter menu?

It was intended to sync the UI menu with the filter in use - this didn't seem to be something automatically handled by ExtJS.

**Comment by JakobMiksch** [comment]
@geographika This issue is solved by #410 
The implementation of `boolean` and `date` is still remaining. 
Maybe they can be added to this issues, otherwise a new issue could be opened for them.

**Comment by geographika** [comment]
Thanks @JakobMiksch - closing. If boolean and date are required I can create a new issue.
